### ISSUE TITLE
Constant error message: "Snooty does not recognize YAML file format"

### ISSUE THREAD

**Issue by CodeMouse92** [vscode-restructuredtext/vscode-restructuredtext#306]
**Environment**
- [x] Extension version:
- [x] Visual Studio Code version: Code 1.57.1 (507ce72a4466fbb27b715c3722558bb15afa9f48, 2021-06-17T13:26:56.255Z)
- [x] Operating System version: Linux x64 5.11.0-7614-generic
- [x] Python version: Python 3.8.5
- [x] sphinx version: 3.5.3 (or 4.0.2)
- [x] A sample of conf.py or its important fragments:

```
  // Just the typical default; this may changes from one project to the next
  // Should also be defined in Workspace Settings
  "restructuredtext.builtDocumentationPath": "${workspaceFolder}/docs/build/html",
  // Point to the restructuredtext-lint executable
  "restructuredtext.linter.executablePath": "restructuredtext-lint",
  // Disable RST linter
  "restructuredtext.linter.run": "off",
  "snooty.languageServerPath": "/home/jason/.venvs/venvdefault/bin/snooty",
  "snooty.sourceFolder": "",
  "snooty.trace.server": "off",
```

**Expected behavior**

Editing a YAML file shouldn't trigger *any* behavior with Snooty or this extension.

**Actual behavior**

Every single time I edit a .yaml file, I get the following error each keypress in the logs. Disabling this extension (restructuredText) resolves the errors.

```
ERROR:pyls_jsonrpc.endpoint:Failed to handle notification textDocument/didOpen: {'textDocument': {'uri': 'file:///home/jason/Canonical/vcs/cpc_jenkins/jenkins/jobs/buildd.yaml', 'languageId': 'yaml', 'version': 1, 'text': '- common_buildd_parameters: &common_buildd_parameters\n    name: \'common_buildd_parameters\'\n    parameters:\n        - string:\n            name: SERIAL\n            description: "The serial of the build to operate on"\n\n- job-template:\n    name: \'{release}-Buildd-Trigger\'\n    node: \'{suite}\'\n    description: |\n        Compare the manifests of the current {suite} images with the archive,\n        and trigger a new image build if any packages have been updated.\n    disabled:\n        !j2: \'{% if configuration["build_triggers_disabled"] %}true{% else %}false{% endif %}\' \n    triggers:\n        - timed: "@daily"\n    builders:\n        - write-out-helper-functions:\n            configuration: \'{obj:configuration}\'\n        - shell:\n            !include-jinja2:\n                - scripts/Swift-Setup.j2\n                - scripts/Should-Build.j2\n        - conditional-step:\n            condition-kind: file-exists\n            condition-filename: "{suite}.trigger"\n            steps:\n                - Set-Current-Build-Description:\n                    description: \'Triggering a new serial\'\n                - trigger-builds:\n                    - project: \'{release}-Buildd-Build-With-Serial\'\n        - conditional-step:\n            condition-kind: not\n            condition-operand:\n                condition-kind: file-exists\n                condition-filename: "{suite}.trigger"\n            steps:\n                - Set-Current-Build-Description:\n                    description: \'No build required\'\n\n- job-template:\n    name: \'{release}-Buildd-Build-With-Serial\'\n    node: \'{suite}\'\n    description: |\n        Determine the next available build serial number to use based on\n        existing builds in swift.\n    builders:\n        - write-out-helper-functions:\n             configuration: \'{obj:configuration}\'\n        - shell:\n            !include-jinja2:\n                - scripts/Swift-Setup.j2\n                - scripts/Determine-Serial.j2\n    publishers:\n        - archive:\n            artifacts: "serial.txt"\n        - trigger-parameterized-builds:\n            - project: \'{release}-Buildd-Build-Images\'\n              property-file: serial.txt\n              fail-on-missing: True\n              condition: SUCCESS\n        - email_notification:\n            configuration: \'{obj:configuration}\'\n        - workspace-cleanup-on-success:\n            release: \'{release}\'\n\n- job-template:\n    project-type: matrix\n    name: \'{release}-Buildd-Build-Images\'\n    description: |\n        This matrix job has a configuration per architecture; each architecture\n        calls on a Launchpad buildd to build images.  Images are downloaded\n        from Launchpad on success and uploaded to swift.\n    <<: *common_buildd_parameters\n    axes:\n        - axis:\n            type: user-defined\n            name: ARCH\n            values:\n                !include: config_includes/arches-Buildd.yaml.inc\n        - axis:\n            type: slave\n            name: NODE\n            values:\n                - \'{suite}\'\n    builders:\n        - write-out-helper-functions:\n             configuration: \'{obj:configuration}\'\n        - write-file:\n            filename: \'mapping.yaml\'\n            content:\n                !include-raw-escape: download_mapping.yaml.inc\n        - Record-Build-Start-In-Swift:\n            configuration: \'{obj:configuration}\'\n            suite: \'{suite}\'\n            image_family: \'{image_family}\'\n        - Build-Images:\n             configuration: \'{obj:configuration}\'\n             suite: \'{suite}\'\n             image_family: \'{image_family}\'\n             image_targets:\n                 - \'all\'\n        - shell:\n            !include-jinja2:\n                - scripts/Swift-Setup.j2\n                - scripts/Store-Images.j2\n    publishers:\n        - archive:\n            artifacts: "out/private/build_log*.txt,out/public/*manifest,out/private/*manifest,out/private/*filelist"\n        - trigger-parameterized-builds:\n            - project: \'{release}-Buildd-Mark-Build-Failed\'\n              current-parameters: True\n              condition: FAILED\n            - project: \'{release}-Buildd-Mark-Build-Complete\'\n              current-parameters: True\n              condition: SUCCESS\n        - email_notification:\n            configuration: \'{obj:configuration}\'\n        - workspace-cleanup-on-success:\n            release: \'{release}\'\n\n- job-template:\n    project-type: matrix\n    name: \'{release}-Buildd-Mark-Build-Complete\'\n    node: \'{suite}\'\n    <<: *common_buildd_parameters\n    axes:\n        - axis:\n            type: user-defined\n            name: ARCH\n            values:\n                !include: config_includes/arches-Buildd.yaml.inc\n    description: |\n        Write metadata to swift indicating Openstack test success and\n        availability of a new daily for in-cloud testing and publication.\n    builders:\n        - write-out-helper-functions:\n            configuration: \'{obj:configuration}\'\n        - Record-Build-Complete-In-Swift:\n            configuration: \'{obj:configuration}\'\n            suite: \'{suite}\'\n            image_family: \'{image_family}\'\n\n- job-template:\n    name: \'{release}-Buildd-Clean-Up-Old-Builds\'\n    node: \'{suite}\'\n    description: |\n        Enforce policies regarding the number of builds per release to keep.\n        Release candidates are not deleted.\n    disabled:\n        !j2: \'{% if configuration["dev_disabled"] %}true{% else %}false{% endif %}\' \n    parameters:\n        - string:\n            name: NUMBER_OF_BUILDS_TO_KEEP\n            default: "{configuration[cleanup][swift][number_of_builds_to_keep]}"\n            description: |\n              Number of builds to keep. All serials with the same base\n              serial (the YYYMMDD part) are counted as one.\n        - string:\n            name: NUMBER_OF_RELEASE_CANDIDATE_BUILDS_TO_KEEP\n            default: "{configuration[cleanup][swift][number_of_release_candidate_builds_to_keep]}"\n            description: |\n              Number of release candidates builds to keep. All serials with\n              the same base serial (the YYYMMDD part) are counted as one.\n    triggers:\n        - timed: "H H * * *"\n    builders:\n        - write-out-helper-functions:\n             configuration: \'{obj:configuration}\'\n        - shell:\n            !include-jinja2:\n                - scripts/Swift-Setup.j2\n                - scripts/Clean-Up-Old-Builds.j2\n\n- job-template:\n    project-type: matrix\n    name: \'{release}-Buildd-Mark-Build-Failed\'\n    node: \'{suite}\'\n    <<: *common_buildd_parameters\n    axes:\n        - axis:\n            type: user-defined\n            name: ARCH\n            values:\n                !include: config_includes/arches-Buildd.yaml.inc\n    description: |\n        Write metadata to swift indicating build failure.\n    builders:\n        - write-out-helper-functions:\n             configuration: \'{obj:configuration}\'\n        - Record-Build-Failed-In-Swift:\n            configuration: \'{obj:configuration}\'\n            suite: \'{suite}\'\n            image_family: \'{image_family}\'\n\n- job-group:\n    name: \'{release}-Buildd-Build-Publish-Jobs\'\n    image_family: \'Buildd\'\n    jobs:\n        - \'{release}-Buildd-Trigger\'\n        - \'{release}-Buildd-Build-With-Serial\'\n        - \'{release}-Buildd-Build-Images\'\n        - \'{release}-Buildd-Mark-Build-Complete\'\n        - \'{release}-Buildd-Clean-Up-Old-Builds\'\n        - \'{release}-Buildd-Mark-Build-Failed\'\n'}}
Traceback (most recent call last):
  File "/home/jason/.local/lib/python3.8/site-packages/pyls_jsonrpc/endpoint.py", line 142, in _handle_notification
    handler_result = handler(params)
  File "/home/jason/.local/lib/python3.8/site-packages/pyls_jsonrpc/dispatchers.py", line 23, in handler
    return method(**(params or {}))
  File "/home/jason/.local/lib/python3.8/site-packages/snooty/language_server.py", line 531, in m_text_document__did_open
    self.update_file(page_path, item.text)
  File "/home/jason/.local/lib/python3.8/site-packages/snooty/language_server.py", line 320, in update_file
    self.project.update(page_path, change)
  File "/home/jason/.local/lib/python3.8/site-packages/snooty/parser.py", line 1566, in update
    self._project.update(path, optional_text)
  File "/home/jason/.local/lib/python3.8/site-packages/snooty/parser.py", line 1313, in update
    raise ValueError("Unknown file type: " + str(path))
ValueError: Unknown file type: /home/jason/code/info.yaml
```

**Detailed Steps to reproduce the problem**

Open and edit a YAML file.

---

Referred from here: https://github.com/microsoft/vscode-python/issues/16583

**Comment by lextm** [comment]
You can disable Snooty alone, https://docs.restructuredtext.net/articles/configuration.html#intellisense

**Comment by CodeMouse92** [comment]
> You can disable Snooty alone, https://docs.restructuredtext.net/articles/configuration.html#intellisense

Weirdly, I *had*, and it still threw that error.

**Comment by lextm** [comment]
Once you change that setting, you need to restart VSCode to let it take effect.

Anyway, I found out how to resolve this, so the fix will be included in release 155.0.0.

**Comment by CodeMouse92** [comment]
Good to hear it's resolved, @lextm. I had restarted VSCode - I've actually had Snooty disabled for over a week.
